projects/core/src/auth/user-auth/facade/auth.service.ts

```diff
@@ -113,8 +113,13 @@ export class AuthService {
    * Returns `true` if the user is logged in; and `false` if the user is anonymous.
    */
   isUserLoggedIn(): Observable<boolean> {
-    return this.authStorageService.getToken().pipe(
-      map((userToken) => Boolean(userToken?.access_token)),
+    return combineLatest([
+      this.authStorageService.getToken(),
+      this.oAuthLibWrapperService.isLoginInProgress(),
```

Comment:
Did you consider any other alternative solutions?

Now `isUserLoggedIn()` indirectly depends now on calling `tryLogin()` specifically.

Nitpicks on details:
- Are there any other cases (besides invoking `tryLogin()`) when the “login process is in progress” and we should allow for observing it via `loginInProgress$`? Currently it’s bound only to the custom implementation of `tryLogin()`. And what about other processes when “login is in progress”? I don’t know TBH, maybe calling another method of `OAuthService`? e.g. `tryLoginWithBlahBlah` etc. or calling the OCC endpoint for login?
- should `isUserLoggedIn()` return false when the “login is in progress”? Or should it hold emitting any values until the login is in progress (e.g. use rxjs `filter("...login is in progress...")`)? And if we hold emiting values, what side effects might it cause?